src/bin/listbox_model.rs

```diff
@@ -0,0 +1,322 @@
+//! # ListBox and ListModel Sample
+//!
+//! This sample demonstrates how to use gtk::Listbox in combination with
+//! gio::ListStore as a model with a custom row type.
+
+#[macro_use]
+extern crate glib;
+extern crate gio;
+extern crate gtk;
+
+extern crate glib_sys as glib_ffi;
+extern crate gobject_sys as gobject_ffi;
+
+extern crate gobject_subclass;
+
+use gio::prelude::*;
+use gtk::prelude::*;
+
+use std::env::args;
+
+use row_data::RowData;
+
+// make moving clones into closures more convenient
+macro_rules! clone {
```

Comment:
We should really add this into a crate or somewhere...

Reply:
clone_army! in the closet crate :)

Reply:
XD